### DISCUSSION TITLE
Constants and parameters

### DISCUSSION THREAD

**Discussion by joaomcteixeira** [THGLab/MCSCE#4] [Implementations]
To my understanding, all constants and parameters needed in the original implementation were hardcoded, right? I mean, there were no routines parsing data from more complex files. If that is the case, I suggest we compile all of those parameters in a single `.py` file that we can have in `mcsce/core/definitions.py` from which the rest of the project will feed.

Do you find this reasonable?

**Comment by JerryJohnsonLee** [comment]
I did not fully walkthrough the lines of code, but my understanding is that part of the parameters are hardcoded (the HPMF part) but other parameters are coming from the standard (amoeba?) force field parameters that is stored in a .prm format file. I am not sure whether we should do the hardcoding, but we do not plan to support different force fields right?

**Reply by joaomcteixeira** [reply]
I believe we do not plan to support different force fields. It is very hard to support different forcefields, we would basically need one parser for each force field.

If the current version uses values in the amoeba force field, then we need to identify the current parser as well.

**Reply by JerryJohnsonLee** [reply]
That is within thinker, and the format of .prm file is very human-readable

**Reply by joaomcteixeira** [reply]
Great! So we just need to identify which `.prm` file in Tinker does the old code read. And we can write a parser ourselves, or hardcode the values from the `.prm`. Can you identify that file @JerryJohnsonLee ? Or all the `prm` files of Tinker, then we will look into them in detail.

**Reply by JerryJohnsonLee** [reply]
I will figure that out. Probably it is the standard amoeba force field parameter set.
